src/utils/env/index.ts

```diff
@@ -1,65 +1,117 @@
 import type { NetlifyAPI } from 'netlify'
 
-import { $TSFixMe } from '../../commands/types.js'
 import { logAndThrowError } from '../command-helpers.js'
 import type { SiteInfo, EnvironmentVariableSource } from '../../utils/types.js'
 
-export const AVAILABLE_CONTEXTS = ['all', 'production', 'deploy-preview', 'branch-deploy', 'dev']
-export const AVAILABLE_SCOPES = ['builds', 'functions', 'runtime', 'post_processing']
-
-type EnvironmentVariableContext = 'all' | 'production' | 'deploy-preview' | 'branch-deploy' | 'dev'
-
-type EnvironmentVariableScope = 'builds' | 'functions' | 'runtime' | 'post_processing'
+/**
+ * Supported values for the user-provided env `context` option.
+ * These all match possible `context` values returned by the Envelope API.
+ * Note that a user may also specify a branch name with the special `branch:my-branch-name` format.
+ */
+export const SUPPORTED_CONTEXTS = ['all', 'production', 'deploy-preview', 'branch-deploy', 'dev'] as const
+/**
+ * Additional aliases for the user-provided env `context` option.
+ */
+const SUPPORTED_CONTEXT_ALIASES = {
+  dp: 'deploy-preview',
+  prod: 'production',
+}
+/**
+ * Supported values for the user-provided env `scope` option.
+ * These exactly match possible `scope` values returned by the Envelope API.
+ * Note that `any` is also supported.
+ */
+export const ALL_ENVELOPE_SCOPES = ['builds', 'functions', 'runtime', 'post_processing'] as const
 
-type EnvironmentVariableValue = {
-  context: EnvironmentVariableContext
+// TODO(serhalp) Netlify API is incorrect - the returned scope is `post_processing`, not `post-processing`
+type EnvelopeEnvVarScope =
+  | Exclude<NonNullable<Awaited<ReturnType<NetlifyAPI['getEnvVars']>>[number]['scopes']>[number], 'post-processing'>
+  | 'post_processing'
+type EnvelopeEnvVar = Awaited<ReturnType<NetlifyAPI['getEnvVars']>>[number] & {
+  scopes: EnvelopeEnvVarScope[]
+}
+type EnvelopeEnvVarContext = NonNullable<EnvelopeEnvVar['values']>[number]['context']
+type EnvelopeEnvVarValue = {
+  /**
+   * The deploy context of the this env var value
+   */
+  context: EnvelopeEnvVarContext
```

Comment:
Possibly related to my point above: In the client's OpenAPI types, this is an optional`?` but here is a required-but-could-be-`undefined` prop. (I'm also a little confused by the inconsistency here--are you inlining the `EnvelopeEnvVarValue` type out of convenience, because it's not possible to pull it out of a return value, or because it needs patching?)

Reply:
Good catch, thanks. I assume that's incorrect in the OpenAPI types, but I'll update it to match.

Reply:
> I'm also a little confused by the inconsistency here--are you inlining the EnvelopeEnvVarValue type out of convenience, because it's not possible to pull it out of a return value, or because it needs patching?

Good question. I guess I'm just redefining the top-level `EnvelopeEnvVarValue` so that I can add describe doc comments on its fields?